### PR TITLE
Adding tests for leaching example

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
The PR adds tests for the leaching example

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: src/prommis/leaching/leach_flowsheet.py [hunk 1]
```diff
@@ -30,137 +18,167 @@
...
+    scaling = TransformationFactory("core.scale_model")
+    scaled_model = scaling.create_using(m, rename=False)
 
-initializer = MSContactorInitializer()
-try:
-    initializer.initialize(scaled_model.fs.leach)
-except:
-    pass
+    # Initialize model
+    # This is likely to fai lto converge, but gives a good enoguh starting point
```
Comment (MarcusHolly) [inline src/prommis/leaching/leach_flowsheet.py:152]:
```suggestion
    # This is likely to fail to converge, but gives a good enough starting point
```

File: src/prommis/leaching/leach_flowsheet.py [hunk 2]
```diff
@@ -30,137 +18,167 @@
...
+        pass
 
-solver = SolverFactory("ipopt")
-solver.solve(scaled_model, tee=True)
+    # Solve scaled model
+    solver = SolverFactory("ipopt")
+    solver.solve(scaled_model, tee=True)
 
-scaling.propagate_solution(scaled_model, m)
+    # Propagate results back ot unscaled model
```
Comment (MarcusHolly) [inline src/prommis/leaching/leach_flowsheet.py:163]:
```suggestion
    # Propagate results back to unscaled model
```

File: src/prommis/leaching/tests/test_leach_flowsheet.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    try:
+        initializer.initialize(scaled_model.fs.leach)
+    except:
+        pass
+
+    # Solve scaled model
+    solver = SolverFactory("ipopt")
+    results = solver.solve(scaled_model, tee=False)
+
+    # Propagate results back ot unscaled model
```
Comment (MarcusHolly) [inline src/prommis/leaching/tests/test_leach_flowsheet.py:58]:
```suggestion
    # Propagate results back to unscaled model
```

**Review by MarcusHolly — APPROVED** [review]
LGTM - just a few typos

**Review by agarciadiego — APPROVED** [review]
LGTM

**Pull request merged**